File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        d = {"I": 1, "V": 5, "X":10, "L": 50,"C": 100, "D": 500, "M": 1000 }
        a = "IVXLCDM"
        result = 0
        for i in range(len(s)):
            sign = s[i]
            if i < len(s) - 1 and a.index(sign) < a.index(s[i + 1]):
                result -= d[sign]
            else:
                result += d[sign]
        return result

File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
        public static int romanToInt(String s) {
            Map<Character, Integer> dict = new HashMap<Character, Integer>();
            int[] val = {1, 5, 10, 50, 100,  500, 1000};
            char[] keys = {'I', 'V', 'X', 'L', 'C', 'D', 'M'};
            String signs = "IVXLCDM";
            for (int i = 0; i <= 6; i++) {
                dict.put(keys[i], val[i]);
            }
            int result = 0;
            for (int i = 0; i < s.length(); i++) {
                char sign = s.charAt(i);
                if (i < s.length() - 1 && signs.indexOf(s.charAt(i + 1)) > signs.indexOf(sign)){
                    result -= dict.get(sign);
                    continue;
                }
                result += dict.get(sign);

            }
            return result;
        }
    }